crates/server/src/store/forwarder/authority.rs

```diff
@@ -51,9 +54,41 @@ impl<P: ConnectionProvider> ForwardAuthority<P> {
 
     /// Read the Authority for the origin from the specified configuration
     pub fn try_from_runtime(
+        origin: Name,
+        zone_type: ZoneType,
+        config: &ForwardConfig,
+        runtime: P,
+    ) -> Result<Self, String> {
+        cfg_if! {
```

Comment:
It's not obvious to me what `cfg_if! {}` is buying us here over simple `#[cfg()]` blocks (and in that case, I prefer the thinner abstraction).